feat(dbmodel): KAM-281: Make source model generation edit-aware

rail-rain opened the pull request:
Running the script causes 400+ changes to `database/`. I inspected some while testing, but there may be some edge cases I missed.

### Changes

Adds a script that reads the DB and update the dot source models to match.

rail-rain committed:
Implement functions to detect changes in tables and columns

rail-rain committed:
Fix errors in logic finding missing and removed tables and columns

rail-rain committed:
Generate columns with extracted constraints

Generate missing docs

Include columns in generated table source models

Setup functions to do manipulations

Make sure markdown files exist for tables for handling columns

Rename functions

Generate missing tables

Fix is_nullable

Clean up data tests generation

Remove deprecated tables

rail-rain committed:
Rename to dbt and sql to clarify the source of data

Fix handleColumns

Draft handling columns

Add notes

Fix markdown removing

Change data types

Don't rewrite ymls and mkds if nothing changed

Handle the case where columns are only missing in the doc or the model

Handle markdowns structurelly

Insert columns sorted

Handle changes in data tests

rail-rain committed:
Remove change detection

It became too complesx for the gain

rail-rain committed:
Include character limits in data_type

rail-rain committed:
Avoid using module wide variables

rail-rain committed:
Refactor

rail-rain committed:
Run on both Tamanu servers and all relevant schemas

Also fix removed tables handling

rail-rain committed:
CHeck for dirty repos

rail-rain committed:
Make dirty repo checking only read ./database/ and also get it check untracked files

rail-rain committed:
Rewrite schema name in case it's wrong

Don't generate doc stabs for columns that has a generic version

Write documents

rail-rain committed:
Delete TODO comments

rail-rain committed:
Fix the lint error

rail-rain committed:
Use while over for

passcod reviewed (changes requested):
just a few minor notes

rail-rain committed:
FIx typos

rail-rain committed:
Fix typos

rail-rain committed:
Import funcrtion by function instead of importing as one module

This changes the code to a CommonJS module from ES6 module. As Lodash is a CommonJS module, this is a more natural way to import them function by function.

rail-rain committed:
Change unmodified let binding to const

rail-rain committed:
Add a better message for the dirty repo error

rail-rain committed:
Extract handlTable into function for better symmetry with handleColumn(s)

rail-rain committed:
Generate descriptions if that was an empty string

rail-rain committed:
Append table names to generated generic description
